From: "Barak A. Pearlmutter" <barak+git@cs.nuim.ie>
Date: Thu, 26 Jan 2012 11:41:32 +0000
Subject: auto gettext

engage autoconf autoheader mechanisms for gettext internationalization

remove '\r' from internationalized strings

remove vobcopy.pot as po/vobcopy.pot is generated by autoreconf

put m4 macros in m4/ subdir
---
 ChangeLog      |   6 +
 Makefile.am    |   5 +-
 configure.ac   |   7 +-
 gettext.h      | 280 ++++++++++++++++++
 m4/ChangeLog   |  10 +
 po/Makevars    |  41 +++
 po/POTFILES.in |   3 +
 vobcopy.c      |  12 +-
 vobcopy.pot    | 873 ---------------------------------------------------------
 9 files changed, 358 insertions(+), 879 deletions(-)
 create mode 100644 ChangeLog
 create mode 100644 gettext.h
 create mode 100644 m4/ChangeLog
 create mode 100644 po/Makevars
 create mode 100644 po/POTFILES.in
 delete mode 100644 vobcopy.pot

diff --git a/ChangeLog b/ChangeLog
new file mode 100644
index 0000000..b33689f
--- /dev/null
+++ b/ChangeLog
@@ -0,0 +1,6 @@
+2012-01-26  gettextize  <bug-gnu-gettext@gnu.org>
+
+	* Makefile.am (SUBDIRS): New variable.
+	(ACLOCAL_AMFLAGS): New variable.
+	(EXTRA_DIST): Add config.rpath, m4/ChangeLog.
+	* configure.ac (AC_CONFIG_FILES): Add po/Makefile.in.
diff --git a/Makefile.am b/Makefile.am
index d5b1296..44b839d 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,3 +1,6 @@
+ACLOCAL_AMFLAGS = -I m4
+SUBDIRS = po
+
 bin_PROGRAMS = vobcopy
 
 vobcopy_SOURCES = dvd.c vobcopy.c dvd.h vobcopy.h
@@ -8,7 +11,7 @@ dist_man_MANS = vobcopy.1
 man1dedir = $(mandir)/de/man1
 man1de_DATA = intl/de/vobcopy.1
 
-EXTRA_DIST = vobcopy.mo vobcopy.pot vobcopy.spec vobcopy.1.de
+EXTRA_DIST = m4/ChangeLog vobcopy.mo vobcopy.pot vobcopy.spec vobcopy.1.de
 
 dist_doc_DATA = alternative_programs.txt Changelog README Release-Notes TODO COPYING
 
diff --git a/configure.ac b/configure.ac
index 8aebe46..62c8773 100644
--- a/configure.ac
+++ b/configure.ac
@@ -4,11 +4,16 @@
 AC_PREREQ([2.67])
 AC_INIT([vobcopy],[1.2.0],[Robos <robos@muon.de>],[vobcopy],[http://vobcopy.org])
 AC_CONFIG_SRCDIR([vobcopy.c])
+AC_CONFIG_MACRO_DIR([m4])
 AC_CONFIG_HEADERS([config.h])
 
 # Enable automake
 AM_INIT_AUTOMAKE([foreign])
 
+# Internationalization support
+AM_GNU_GETTEXT([external])
+AM_GNU_GETTEXT_VERSION([0.18])
+
 # Checks for programs.
 AC_PROG_CC
 AC_PROG_LN_S
@@ -33,5 +38,5 @@ AC_FUNC_MALLOC
 AC_CHECK_FUNCS([alarm fdatasync getcwd getmntent getmntinfo memset mkdir setlocale strcasecmp strerror strpbrk strstr])
 
 # trailer
-AC_CONFIG_FILES([Makefile])
+AC_CONFIG_FILES([Makefile po/Makefile.in])
 AC_OUTPUT
diff --git a/gettext.h b/gettext.h
new file mode 100644
index 0000000..e76b592
--- /dev/null
+++ b/gettext.h
@@ -0,0 +1,280 @@
+/* Convenience header for conditional use of GNU <libintl.h>.
+   Copyright (C) 1995-1998, 2000-2002, 2004-2006, 2009 Free Software Foundation, Inc.
+
+   This program is free software; you can redistribute it and/or modify it
+   under the terms of the GNU General Public License as published
+   by the Free Software Foundation; either version 3, or (at your option)
+   any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Library General Public License for more details.
+
+   You should have received a copy of the GNU General Public
+   License along with this program; if not, write to the Free Software
+   Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301,
+   USA.  */
+
+#ifndef _LIBGETTEXT_H
+#define _LIBGETTEXT_H 1
+
+/* NLS can be disabled through the configure --disable-nls option.  */
+#if ENABLE_NLS
+
+/* Get declarations of GNU message catalog functions.  */
+# include <libintl.h>
+
+/* You can set the DEFAULT_TEXT_DOMAIN macro to specify the domain used by
+   the gettext() and ngettext() macros.  This is an alternative to calling
+   textdomain(), and is useful for libraries.  */
+# ifdef DEFAULT_TEXT_DOMAIN
+#  undef gettext
+#  define gettext(Msgid) \
+     dgettext (DEFAULT_TEXT_DOMAIN, Msgid)
+#  undef ngettext
+#  define ngettext(Msgid1, Msgid2, N) \
+     dngettext (DEFAULT_TEXT_DOMAIN, Msgid1, Msgid2, N)
+# endif
+
+#else
+
+/* Solaris /usr/include/locale.h includes /usr/include/libintl.h, which
+   chokes if dcgettext is defined as a macro.  So include it now, to make
+   later inclusions of <locale.h> a NOP.  We don't include <libintl.h>
+   as well because people using "gettext.h" will not include <libintl.h>,
+   and also including <libintl.h> would fail on SunOS 4, whereas <locale.h>
+   is OK.  */
+#if defined(__sun)
+# include <locale.h>
+#endif
+
+/* Many header files from the libstdc++ coming with g++ 3.3 or newer include
+   <libintl.h>, which chokes if dcgettext is defined as a macro.  So include
+   it now, to make later inclusions of <libintl.h> a NOP.  */
+#if defined(__cplusplus) && defined(__GNUG__) && (__GNUC__ >= 3)
+# include <cstdlib>
+# if (__GLIBC__ >= 2) || _GLIBCXX_HAVE_LIBINTL_H
+#  include <libintl.h>
+# endif
+#endif
+
+/* Disabled NLS.
+   The casts to 'const char *' serve the purpose of producing warnings
+   for invalid uses of the value returned from these functions.
+   On pre-ANSI systems without 'const', the config.h file is supposed to
+   contain "#define const".  */
+# undef gettext
+# define gettext(Msgid) ((const char *) (Msgid))
+# undef dgettext
+# define dgettext(Domainname, Msgid) ((void) (Domainname), gettext (Msgid))
+# undef dcgettext
+# define dcgettext(Domainname, Msgid, Category) \
+    ((void) (Category), dgettext (Domainname, Msgid))
+# undef ngettext
+# define ngettext(Msgid1, Msgid2, N) \
+    ((N) == 1 \
+     ? ((void) (Msgid2), (const char *) (Msgid1)) \
+     : ((void) (Msgid1), (const char *) (Msgid2)))
+# undef dngettext
+# define dngettext(Domainname, Msgid1, Msgid2, N) \
+    ((void) (Domainname), ngettext (Msgid1, Msgid2, N))
+# undef dcngettext
+# define dcngettext(Domainname, Msgid1, Msgid2, N, Category) \
+    ((void) (Category), dngettext(Domainname, Msgid1, Msgid2, N))
+# undef textdomain
+# define textdomain(Domainname) ((const char *) (Domainname))
+# undef bindtextdomain
+# define bindtextdomain(Domainname, Dirname) \
+    ((void) (Domainname), (const char *) (Dirname))
+# undef bind_textdomain_codeset
+# define bind_textdomain_codeset(Domainname, Codeset) \
+    ((void) (Domainname), (const char *) (Codeset))
+
+#endif
+
+/* A pseudo function call that serves as a marker for the automated
+   extraction of messages, but does not call gettext().  The run-time
+   translation is done at a different place in the code.
+   The argument, String, should be a literal string.  Concatenated strings
+   and other string expressions won't work.
+   The macro's expansion is not parenthesized, so that it is suitable as
+   initializer for static 'char[]' or 'const char[]' variables.  */
+#define gettext_noop(String) String
+
+/* The separator between msgctxt and msgid in a .mo file.  */
+#define GETTEXT_CONTEXT_GLUE "\004"
+
+/* Pseudo function calls, taking a MSGCTXT and a MSGID instead of just a
+   MSGID.  MSGCTXT and MSGID must be string literals.  MSGCTXT should be
+   short and rarely need to change.
+   The letter 'p' stands for 'particular' or 'special'.  */
+#ifdef DEFAULT_TEXT_DOMAIN
+# define pgettext(Msgctxt, Msgid) \
+   pgettext_aux (DEFAULT_TEXT_DOMAIN, Msgctxt GETTEXT_CONTEXT_GLUE Msgid, Msgid, LC_MESSAGES)
+#else
+# define pgettext(Msgctxt, Msgid) \
+   pgettext_aux (NULL, Msgctxt GETTEXT_CONTEXT_GLUE Msgid, Msgid, LC_MESSAGES)
+#endif
+#define dpgettext(Domainname, Msgctxt, Msgid) \
+  pgettext_aux (Domainname, Msgctxt GETTEXT_CONTEXT_GLUE Msgid, Msgid, LC_MESSAGES)
+#define dcpgettext(Domainname, Msgctxt, Msgid, Category) \
+  pgettext_aux (Domainname, Msgctxt GETTEXT_CONTEXT_GLUE Msgid, Msgid, Category)
+#ifdef DEFAULT_TEXT_DOMAIN
+# define npgettext(Msgctxt, Msgid, MsgidPlural, N) \
+   npgettext_aux (DEFAULT_TEXT_DOMAIN, Msgctxt GETTEXT_CONTEXT_GLUE Msgid, Msgid, MsgidPlural, N, LC_MESSAGES)
+#else
+# define npgettext(Msgctxt, Msgid, MsgidPlural, N) \
+   npgettext_aux (NULL, Msgctxt GETTEXT_CONTEXT_GLUE Msgid, Msgid, MsgidPlural, N, LC_MESSAGES)
+#endif
+#define dnpgettext(Domainname, Msgctxt, Msgid, MsgidPlural, N) \
+  npgettext_aux (Domainname, Msgctxt GETTEXT_CONTEXT_GLUE Msgid, Msgid, MsgidPlural, N, LC_MESSAGES)
+#define dcnpgettext(Domainname, Msgctxt, Msgid, MsgidPlural, N, Category) \
+  npgettext_aux (Domainname, Msgctxt GETTEXT_CONTEXT_GLUE Msgid, Msgid, MsgidPlural, N, Category)
+
+#ifdef __GNUC__
+__inline
+#else
+#ifdef __cplusplus
+inline
+#endif
+#endif
+static const char *
+pgettext_aux (const char *domain,
+              const char *msg_ctxt_id, const char *msgid,
+              int category)
+{
+  const char *translation = dcgettext (domain, msg_ctxt_id, category);
+  if (translation == msg_ctxt_id)
+    return msgid;
+  else
+    return translation;
+}
+
+#ifdef __GNUC__
+__inline
+#else
+#ifdef __cplusplus
+inline
+#endif
+#endif
+static const char *
+npgettext_aux (const char *domain,
+               const char *msg_ctxt_id, const char *msgid,
+               const char *msgid_plural, unsigned long int n,
+               int category)
+{
+  const char *translation =
+    dcngettext (domain, msg_ctxt_id, msgid_plural, n, category);
+  if (translation == msg_ctxt_id || translation == msgid_plural)
+    return (n == 1 ? msgid : msgid_plural);
+  else
+    return translation;
+}
+
+/* The same thing extended for non-constant arguments.  Here MSGCTXT and MSGID
+   can be arbitrary expressions.  But for string literals these macros are
+   less efficient than those above.  */
+
+#include <string.h>
+
+#define _LIBGETTEXT_HAVE_VARIABLE_SIZE_ARRAYS \
+  (((__GNUC__ >= 3 || __GNUG__ >= 2) && !__STRICT_ANSI__) \
+   /* || __STDC_VERSION__ >= 199901L */ )
+
+#if !_LIBGETTEXT_HAVE_VARIABLE_SIZE_ARRAYS
+#include <stdlib.h>
+#endif
+
+#define pgettext_expr(Msgctxt, Msgid) \
+  dcpgettext_expr (NULL, Msgctxt, Msgid, LC_MESSAGES)
+#define dpgettext_expr(Domainname, Msgctxt, Msgid) \
+  dcpgettext_expr (Domainname, Msgctxt, Msgid, LC_MESSAGES)
+
+#ifdef __GNUC__
+__inline
+#else
+#ifdef __cplusplus
+inline
+#endif
+#endif
+static const char *
+dcpgettext_expr (const char *domain,
+                 const char *msgctxt, const char *msgid,
+                 int category)
+{
+  size_t msgctxt_len = strlen (msgctxt) + 1;
+  size_t msgid_len = strlen (msgid) + 1;
+  const char *translation;
+#if _LIBGETTEXT_HAVE_VARIABLE_SIZE_ARRAYS
+  char msg_ctxt_id[msgctxt_len + msgid_len];
+#else
+  char buf[1024];
+  char *msg_ctxt_id =
+    (msgctxt_len + msgid_len <= sizeof (buf)
+     ? buf
+     : (char *) malloc (msgctxt_len + msgid_len));
+  if (msg_ctxt_id != NULL)
+#endif
+    {
+      memcpy (msg_ctxt_id, msgctxt, msgctxt_len - 1);
+      msg_ctxt_id[msgctxt_len - 1] = '\004';
+      memcpy (msg_ctxt_id + msgctxt_len, msgid, msgid_len);
+      translation = dcgettext (domain, msg_ctxt_id, category);
+#if !_LIBGETTEXT_HAVE_VARIABLE_SIZE_ARRAYS
+      if (msg_ctxt_id != buf)
+        free (msg_ctxt_id);
+#endif
+      if (translation != msg_ctxt_id)
+        return translation;
+    }
+  return msgid;
+}
+
+#define npgettext_expr(Msgctxt, Msgid, MsgidPlural, N) \
+  dcnpgettext_expr (NULL, Msgctxt, Msgid, MsgidPlural, N, LC_MESSAGES)
+#define dnpgettext_expr(Domainname, Msgctxt, Msgid, MsgidPlural, N) \
+  dcnpgettext_expr (Domainname, Msgctxt, Msgid, MsgidPlural, N, LC_MESSAGES)
+
+#ifdef __GNUC__
+__inline
+#else
+#ifdef __cplusplus
+inline
+#endif
+#endif
+static const char *
+dcnpgettext_expr (const char *domain,
+                  const char *msgctxt, const char *msgid,
+                  const char *msgid_plural, unsigned long int n,
+                  int category)
+{
+  size_t msgctxt_len = strlen (msgctxt) + 1;
+  size_t msgid_len = strlen (msgid) + 1;
+  const char *translation;
+#if _LIBGETTEXT_HAVE_VARIABLE_SIZE_ARRAYS
+  char msg_ctxt_id[msgctxt_len + msgid_len];
+#else
+  char buf[1024];
+  char *msg_ctxt_id =
+    (msgctxt_len + msgid_len <= sizeof (buf)
+     ? buf
+     : (char *) malloc (msgctxt_len + msgid_len));
+  if (msg_ctxt_id != NULL)
+#endif
+    {
+      memcpy (msg_ctxt_id, msgctxt, msgctxt_len - 1);
+      msg_ctxt_id[msgctxt_len - 1] = '\004';
+      memcpy (msg_ctxt_id + msgctxt_len, msgid, msgid_len);
+      translation = dcngettext (domain, msg_ctxt_id, msgid_plural, n, category);
+#if !_LIBGETTEXT_HAVE_VARIABLE_SIZE_ARRAYS
+      if (msg_ctxt_id != buf)
+        free (msg_ctxt_id);
+#endif
+      if (!(translation == msg_ctxt_id || translation == msgid_plural))
+        return translation;
+    }
+  return (n == 1 ? msgid : msgid_plural);
+}
+
+#endif /* _LIBGETTEXT_H */
diff --git a/m4/ChangeLog b/m4/ChangeLog
new file mode 100644
index 0000000..2e2bd44
--- /dev/null
+++ b/m4/ChangeLog
@@ -0,0 +1,10 @@
+2012-01-26  gettextize  <bug-gnu-gettext@gnu.org>
+
+	* gettext.m4: New file, from gettext-0.18.1.
+	* iconv.m4: New file, from gettext-0.18.1.
+	* lib-ld.m4: New file, from gettext-0.18.1.
+	* lib-link.m4: New file, from gettext-0.18.1.
+	* lib-prefix.m4: New file, from gettext-0.18.1.
+	* nls.m4: New file, from gettext-0.18.1.
+	* po.m4: New file, from gettext-0.18.1.
+	* progtest.m4: New file, from gettext-0.18.1.
diff --git a/po/Makevars b/po/Makevars
new file mode 100644
index 0000000..32692ab
--- /dev/null
+++ b/po/Makevars
@@ -0,0 +1,41 @@
+# Makefile variables for PO directory in any package using GNU gettext.
+
+# Usually the message domain is the same as the package name.
+DOMAIN = $(PACKAGE)
+
+# These two variables depend on the location of this directory.
+subdir = po
+top_builddir = ..
+
+# These options get passed to xgettext.
+XGETTEXT_OPTIONS = --keyword=_ --keyword=N_
+
+# This is the copyright holder that gets inserted into the header of the
+# $(DOMAIN).pot file.  Set this to the copyright holder of the surrounding
+# package.  (Note that the msgstr strings, extracted from the package's
+# sources, belong to the copyright holder of the package.)  Translators are
+# expected to transfer the copyright for their translations to this person
+# or entity, or to disclaim their copyright.  The empty string stands for
+# the public domain; in this case the translators are expected to disclaim
+# their copyright.
+COPYRIGHT_HOLDER = Free Software Foundation, Inc.
+
+# This is the email address or URL to which the translators shall report
+# bugs in the untranslated strings:
+# - Strings which are not entire sentences, see the maintainer guidelines
+#   in the GNU gettext documentation, section 'Preparing Strings'.
+# - Strings which use unclear terms or require additional context to be
+#   understood.
+# - Strings which make invalid assumptions about notation of date, time or
+#   money.
+# - Pluralisation problems.
+# - Incorrect English spelling.
+# - Incorrect formatting.
+# It can be your email address, or a mailing list address where translators
+# can write to without being subscribed, or the URL of a web page through
+# which the translators can contact you.
+MSGID_BUGS_ADDRESS =
+
+# This is the list of locale categories, beyond LC_MESSAGES, for which the
+# message catalogs shall be used.  It is usually empty.
+EXTRA_LOCALE_CATEGORIES =
diff --git a/po/POTFILES.in b/po/POTFILES.in
new file mode 100644
index 0000000..2dbd919
--- /dev/null
+++ b/po/POTFILES.in
@@ -0,0 +1,3 @@
+# List of source files which contain translatable strings.
+dvd.c
+vobcopy.c
diff --git a/vobcopy.c b/vobcopy.c
index 9d165f6..63aa21d 100644
--- a/vobcopy.c
+++ b/vobcopy.c
@@ -1196,8 +1196,9 @@ next: /*for the goto - ugly, I know... */
                         }
                       /* progress indicator */
                       tmp_i = i;
-                      fprintf( stderr, _("%4.0fkB of %4.0fkB written\r"),
+                      fprintf( stderr, _("%4.0fkB of %4.0fkB written"),
                                ( tmp_i+1 )*( DVD_VIDEO_LB_LEN/1024 ), tmp_file_size/1024 );
+                      fprintf( stderr, "\r" );
                     }
                   fprintf( stderr, _("\n"));
                   if( !stdout_flag )
@@ -1231,8 +1232,9 @@ next: /*for the goto - ugly, I know... */
                         }
                       /* progress indicator */
                       tmp_i = i;
-                      fprintf( stderr, _("%4.0fkB of %4.0fkB written\r"),
+                      fprintf( stderr, _("%4.0fkB of %4.0fkB written"),
                                ( tmp_i+1 )*( DVD_VIDEO_LB_LEN/1024 ), tmp_file_size/1024 );
+                      fprintf( stderr, "\r");
                     }
                   fprintf( stderr, _("\n"));
                   if( !stdout_flag )
@@ -1356,7 +1358,8 @@ next: /*for the goto - ugly, I know... */
                           tmp_i = ( i-start );
 
                           percent = ( ( ( ( tmp_i+1 )*DVD_VIDEO_LB_LEN )*100 )/tmp_file_size );
-                          fprintf( stderr, _("\r%4.0fMB of %4.0fMB written "),
+                          fprintf( stderr, "\r");
+                          fprintf( stderr, _("%4.0fMB of %4.0fMB written "),
                                    ( ( tmp_i+1 )*DVD_VIDEO_LB_LEN )/( 1024*1024 ),
                                    ( tmp_file_size+2048 )/( 1024*1024 ) );
                           fprintf( stderr, _("( %3.1f %% ) "), percent );
@@ -1365,7 +1368,8 @@ next: /*for the goto - ugly, I know... */
                     }
 /*this is just so that at the end it actually says 100.0% all the time... */
 /*TODO: if it is correct to always assume it's 100% is a good question.... */
-/*                  fprintf( stderr, _("\r%4.0fMB of %4.0fMB written "),
+/*                  fprintf( stderr, "\r");
+                  fprintf( stderr, _("%4.0fMB of %4.0fMB written "),
                            ( ( tmp_i+1 )*DVD_VIDEO_LB_LEN )/( 1024*1024 ),
                            ( tmp_file_size+2048 )/( 1024*1024 ) );
                   fprintf( stderr, _("( 100.0%% ) ") );
diff --git a/vobcopy.pot b/vobcopy.pot
deleted file mode 100644
index 2340029..0000000
--- a/vobcopy.pot
+++ /dev/null
@@ -1,873 +0,0 @@
-# SOME DESCRIPTIVE TITLE.
-# Copyright (C) YEAR THE PACKAGE'S COPYRIGHT HOLDER
-# This file is distributed under the same license as the PACKAGE package.
-# FIRST AUTHOR <EMAIL@ADDRESS>, YEAR.
-#
-#, fuzzy
-msgid ""
-msgstr ""
-"Project-Id-Version: PACKAGE VERSION\n"
-"Report-Msgid-Bugs-To: \n"
-"POT-Creation-Date: 2008-03-31 23:48+0200\n"
-"PO-Revision-Date: YEAR-MO-DA HO:MI+ZONE\n"
-"Last-Translator: FULL NAME <EMAIL@ADDRESS>\n"
-"Language-Team: LANGUAGE <LL@li.org>\n"
-"MIME-Version: 1.0\n"
-"Content-Type: text/plain; charset=CHARSET\n"
-"Content-Transfer-Encoding: 8bit\n"
-
-#: vobcopy.c:189
-#, c-format
-msgid "[Error] The thing behind -a has to be a number! \n"
-msgstr ""
-
-#: vobcopy.c:196
-#, c-format
-msgid "[Hint] Um, you set angle to 0, try 1 instead ;-)\n"
-msgstr ""
-
-#: vobcopy.c:204
-#, c-format
-msgid "[Error] The thing behind -b has to be a number! \n"
-msgstr ""
-
-#: vobcopy.c:211 vobcopy.c:234 vobcopy.c:265 vobcopy.c:287
-#, c-format
-msgid "[Error] Wrong suffix behind -b, only b,k,m or g \n"
-msgstr ""
-
-#: vobcopy.c:248
-#, c-format
-msgid "[Error] The thing behind -c has to be a number! \n"
-msgstr ""
-
-#: vobcopy.c:258
-#, c-format
-msgid "[Error] The thing behind -e has to be a number! \n"
-msgstr ""
-
-#: vobcopy.c:310
-#, c-format
-msgid "[Error] Erm, the number comes behind -n ... \n"
-msgstr ""
-
-#: vobcopy.c:313
-#, c-format
-msgid ""
-"[Hint] You use -i. Normally this is not necessary, vobcopy finds the input "
-"dir by itself. This option is only there if vobcopy makes trouble.\n"
-msgstr ""
-
-#: vobcopy.c:314
-#, c-format
-msgid ""
-"[Hint] If vobcopy makes trouble, please mail me so that I can fix this "
-"(robos@muon.de). Thanks\n"
-msgstr ""
-
-#: vobcopy.c:318
-#, c-format
-msgid ""
-"[Error] Please don't use -i /dev/something in this version, only the next "
-"version will support this again.\n"
-msgstr ""
-
-#: vobcopy.c:319
-#, c-format
-msgid ""
-"[Hint] Please use the mount point instead (/cdrom, /dvd, /mnt/dvd or "
-"something)\n"
-msgstr ""
-
-#: vobcopy.c:340
-#, c-format
-msgid "[Error] The thing behind -n has to be a number! \n"
-msgstr ""
-
-#: vobcopy.c:350 vobcopy.c:382
-#, c-format
-msgid "[Hint] Erm, the number comes behind -n ... \n"
-msgstr ""
-
-#: vobcopy.c:376
-#, c-format
-msgid "[Error] Please specify output dirs in this order: -o -1 -2 -3 -4 \n"
-msgstr ""
-
-#: vobcopy.c:410
-#, c-format
-msgid "[Error] The thing behind -F has to be a number! \n"
-msgstr ""
-
-#: vobcopy.c:416
-#, c-format
-msgid ""
-"[Hint] The largest value for -F is %d at the moment - used that one...\n"
-msgstr ""
-
-#: vobcopy.c:427
-#, c-format
-msgid "[Error] The thing behind -w has to be a number! \n"
-msgstr ""
-
-#: vobcopy.c:433
-#, c-format
-msgid "[Hint] Negative minutes aren't allowed - disabling watchdog.\n"
-msgstr ""
-
-#: vobcopy.c:479 vobcopy.c:493
-#, c-format
-msgid "[Error] Wrong option.\n"
-msgstr ""
-
-#: vobcopy.c:486
-#, c-format
-msgid "[Error] %s was compiled without support for long options.\n"
-msgstr ""
-
-#: vobcopy.c:499 vobcopy.c:2022
-#, c-format
-msgid "Vobcopy "
-msgstr ""
-
-#: vobcopy.c:500
-#, c-format
-msgid ""
-"[Hint] All lines starting with \"libdvdread:\" are not from vobcopy but from "
-"the libdvdread-library\n"
-msgstr ""
-
-#: vobcopy.c:511
-#, c-format
-msgid ""
-"\n"
-"[Error] Hmm, the path length of your current directory is really large "
-"(>255)\n"
-msgstr ""
-
-#: vobcopy.c:512
-#, c-format
-msgid "[Hint] Change to a path with shorter path length pleeeease ;-)\n"
-msgstr ""
-
-#: vobcopy.c:526
-#, c-format
-msgid "[Hint] Quiet mode - All messages will now end up in %s\n"
-msgstr ""
-
-#: vobcopy.c:575
-#, c-format
-msgid "[Info] Uhu, super-verbose\n"
-msgstr ""
-
-#: vobcopy.c:615
-#, c-format
-msgid "[Info] The log-file is written to %s\n"
-msgstr ""
-
-#: vobcopy.c:616
-#, c-format
-msgid ""
-"[Hint] Make sure that vobcopy doesn't have to ask questions (like "
-"overwriting of old files), these questions end up in the log file so you "
-"don't see them...\n"
-msgstr ""
-
-#: vobcopy.c:617
-#, c-format
-msgid ""
-"[Hint] If you don't like that position, use -L /path/to/logfile/ instead of -"
-"v -v\n"
-msgstr ""
-
-#: vobcopy.c:633
-#, c-format
-msgid "--------------------------------------------------------------------------------\n"
-msgstr ""
-
-#: vobcopy.c:634
-#, c-format
-msgid "[Info] Called: %s\n"
-msgstr ""
-
-#: vobcopy.c:640
-#, c-format
-msgid ""
-"\n"
-"[Error] There can be only one: either -m or -n...'\n"
-msgstr ""
-
-#: vobcopy.c:653
-#, c-format
-msgid ""
-"\n"
-"[Error] Bloody path to long '%s'\n"
-msgstr ""
-
-#: vobcopy.c:665
-#, c-format
-msgid "[Error] Could not get the device which belongs to the given path!\n"
-msgstr ""
-
-#: vobcopy.c:678
-#, c-format
-msgid "[Error] Could not get the device and path! Maybe not mounted the dvd?\n"
-msgstr ""
-
-#: vobcopy.c:688
-#, c-format
-msgid ""
-"\n"
-"[Info] Path to dvd: %s\n"
-msgstr ""
-
-#: vobcopy.c:692
-#, c-format
-msgid ""
-"\n"
-"[Error] Path thingy didn't work '%s'\n"
-msgstr ""
-
-#: vobcopy.c:693
-#, c-format
-msgid "[Error] Try something like -i /cdrom, /dvd  or /mnt/dvd \n"
-msgstr ""
-
-#: vobcopy.c:695
-#, c-format
-msgid ""
-"[Hint] By the way, you have %i cdroms|dvds mounted, that probably caused the "
-"problem\n"
-msgstr ""
-
-#: vobcopy.c:706
-#, c-format
-msgid "[Info] Name of the dvd: %s\n"
-msgstr ""
-
-#: vobcopy.c:722
-#, c-format
-msgid "[Error] Can't open VMG info.\n"
-msgstr ""
-
-#: vobcopy.c:754
-#, c-format
-msgid "[Info] There are %d titles on this DVD.\n"
-msgstr ""
-
-#: vobcopy.c:758
-#, c-format
-msgid "[Error] Invalid title %d.\n"
-msgstr ""
-
-#: vobcopy.c:769
-#, c-format
-msgid "[Info] There are %i chapters on the dvd.\n"
-msgstr ""
-
-#: vobcopy.c:770
-#, c-format
-msgid "[Info] Most chapters has title %i with %d chapters.\n"
-msgstr ""
-
-#: vobcopy.c:775 vobcopy.c:818
-#, c-format
-msgid "[Info] All titles:\n"
-msgstr ""
-
-#: vobcopy.c:780
-#, c-format
-msgid "[Info] Title %i has %d chapters.\n"
-msgstr ""
-
-#: vobcopy.c:783
-#, c-format
-msgid "[Info] Title %i has %d chapter.\n"
-msgstr ""
-
-#: vobcopy.c:793
-#, c-format
-msgid "[Error] Invalid chapter %d\n"
-msgstr ""
-
-#: vobcopy.c:807
-#, c-format
-msgid ""
-"\n"
-"[Info] There are %d angles on this dvd.\n"
-msgstr ""
-
-#: vobcopy.c:810
-#, c-format
-msgid "[Error] Invalid angle %d\n"
-msgstr ""
-
-#: vobcopy.c:823
-#, c-format
-msgid "[Info] Title %i has %d angles.\n"
-msgstr ""
-
-#: vobcopy.c:826
-#, c-format
-msgid "[Info] Title %i has %d angle.\n"
-msgstr ""
-
-#: vobcopy.c:844
-#, c-format
-msgid ""
-"\n"
-"[Error] Something went wrong during the size detection of the"
-msgstr ""
-
-#: vobcopy.c:845
-#, c-format
-msgid ""
-"\n"
-"[Error] vobs, size check at the end won't work (probably), but I continue "
-"anyway\n"
-"\n"
-msgstr ""
-
-#: vobcopy.c:874
-#, c-format
-msgid ""
-"\n"
-"[Info] Setting watchdog timer to %d minutes\n"
-msgstr ""
-
-#: vobcopy.c:884 vobcopy.c:1364 vobcopy.c:1444 vobcopy.c:1514
-#, c-format
-msgid ""
-"\n"
-"[Info] DVD-name: %s\n"
-msgstr ""
-
-#: vobcopy.c:885 vobcopy.c:1445
-#, c-format
-msgid "[Info]  Disk free: %.0f MB\n"
-msgstr ""
-
-#: vobcopy.c:886 vobcopy.c:1447
-#, c-format
-msgid "[Info]  Vobs size: %.0f MB\n"
-msgstr ""
-
-#: vobcopy.c:911
-#, c-format
-msgid "[Info] Writing files to this dir: %s\n"
-msgstr ""
-
-#: vobcopy.c:924
-#, c-format
-msgid ""
-"[Error] Hmm, weird, the dir video_ts|VIDEO_TS on the dvd couldn't be opened\n"
-msgstr ""
-
-#: vobcopy.c:925
-#, c-format
-msgid "[Error] The dir to be opened was: %s\n"
-msgstr ""
-
-#: vobcopy.c:926
-#, c-format
-msgid "[Hint] Please mail me what your vobcopy call plus -v -v spits out\n"
-msgstr ""
-
-#: vobcopy.c:1001
-#, c-format
-msgid ""
-"\n"
-"[Hint] File on dvd ends in \";?\" (%s)\n"
-msgstr ""
-
-#: vobcopy.c:1009
-#, c-format
-msgid "[Info] Writing to %s \n"
-msgstr ""
-
-#: vobcopy.c:1014
-#, c-format
-msgid ""
-"\n"
-"[Error] File '%s' already exists, [o]verwrite, [s]kip or [q]uit?  "
-msgstr ""
-
-#: vobcopy.c:1026 vobcopy.c:1071 vobcopy.c:1097 vobcopy.c:1610 vobcopy.c:1661
-#: vobcopy.c:1677 vobcopy.c:1707
-#, c-format
-msgid ""
-"\n"
-"[Error] Error opening file %s\n"
-msgstr ""
-
-#: vobcopy.c:1027 vobcopy.c:1072 vobcopy.c:1098 vobcopy.c:1141 vobcopy.c:1176
-#: vobcopy.c:1611 vobcopy.c:1750
-#, c-format
-msgid "[Error] Error: %s\n"
-msgstr ""
-
-#: vobcopy.c:1048
-#, c-format
-msgid ""
-"\n"
-"[Hint] Please choose [o]verwrite, [s]kip, or [q]uit the next time ;-)\n"
-msgstr ""
-
-#: vobcopy.c:1059 vobcopy.c:1599
-#, c-format
-msgid ""
-"\n"
-"[Error] File '%s' already exists, [o]verwrite or [q]uit? \n"
-msgstr ""
-
-#: vobcopy.c:1088
-#, c-format
-msgid ""
-"\n"
-"[Hint] Please choose [o]verwrite or [q]uit the next time ;-)\n"
-msgstr ""
-
-#: vobcopy.c:1140 vobcopy.c:1175 vobcopy.c:1287
-#, c-format
-msgid ""
-"\n"
-"[Error] Error writing to %s \n"
-msgstr ""
-
-#: vobcopy.c:1146 vobcopy.c:1181
-#, c-format
-msgid "%4.0fkB of %4.0fkB written\r"
-msgstr ""
-
-#: vobcopy.c:1149 vobcopy.c:1184 vobcopy.c:1317 vobcopy.c:1720
-#, c-format
-msgid "\n"
-msgstr ""
-
-#: vobcopy.c:1154 vobcopy.c:1189 vobcopy.c:1322 vobcopy.c:1773
-#, c-format
-msgid ""
-"\n"
-"[Error] error writing to %s \n"
-msgstr ""
-
-#: vobcopy.c:1155 vobcopy.c:1190 vobcopy.c:1323 vobcopy.c:1774 vobcopy.c:2144
-#, c-format
-msgid "[Error] error: %s\n"
-msgstr ""
-
-#: vobcopy.c:1237
-#, c-format
-msgid "[Info] Can't stat() %s.\n"
-msgstr ""
-
-#: vobcopy.c:1243
-#, c-format
-msgid "[Info] Vob %d %d (%s) has a size of %lli\n"
-msgstr ""
-
-#: vobcopy.c:1257
-#, c-format
-msgid "[Info] Start of %s at %d blocks \n"
-msgstr ""
-
-#: vobcopy.c:1279
-#, c-format
-msgid "[Warn] Had to skip %d blocks!  "
-msgstr ""
-
-#: vobcopy.c:1288
-#, c-format
-msgid "[Error] Error: %s, errno: %d \n"
-msgstr ""
-
-#: vobcopy.c:1301 vobcopy.c:1311
-#, c-format
-msgid "%4.0fMB of %4.0fMB written "
-msgstr ""
-
-#: vobcopy.c:1304
-#, c-format
-msgid "( %3.1f %% ) \r"
-msgstr ""
-
-#: vobcopy.c:1314
-#, c-format
-msgid "( 100.0%% ) \r"
-msgstr ""
-
-#: vobcopy.c:1340
-#, c-format
-msgid ""
-"[Error] Not enough free space on the destination dir. Please choose another "
-"one or -f\n"
-msgstr ""
-
-#: vobcopy.c:1341
-#, c-format
-msgid "[Error] or dirs behind -1, -2 ... are NOT allowed with -m!\n"
-msgstr ""
-
-#: vobcopy.c:1356
-#, c-format
-msgid "[Info] Using Title: %i\n"
-msgstr ""
-
-#: vobcopy.c:1357
-#, c-format
-msgid "[Info] Title has %d chapters and %d angles\n"
-msgstr ""
-
-#: vobcopy.c:1358
-#, c-format
-msgid "[Info] Using Chapter: %i\n"
-msgstr ""
-
-#: vobcopy.c:1359
-#, c-format
-msgid "[Info] Using Angle: %i\n"
-msgstr ""
-
-#: vobcopy.c:1365
-#, c-format
-msgid "[Info]  Disk free: %f MB\n"
-msgstr ""
-
-#: vobcopy.c:1366
-#, c-format
-msgid "[Info]  Vobs size: %f MB\n"
-msgstr ""
-
-#: vobcopy.c:1381
-#, c-format
-msgid "[Error] Can't open the title %d info file.\n"
-msgstr ""
-
-#: vobcopy.c:1408
-#, c-format
-msgid "[Error] Can't open title VOBS (VTS_%02d_1.VOB).\n"
-msgstr ""
-
-#: vobcopy.c:1425
-#, c-format
-msgid "[Info] Vob_size was 0\n"
-msgstr ""
-
-#: vobcopy.c:1432
-#, c-format
-msgid ""
-"\n"
-"[Info] Difference between vobsize read from cd and size returned from "
-"libdvdread:\n"
-msgstr ""
-
-#: vobcopy.c:1437
-#, c-format
-msgid ""
-"[Info] Vob_size (stat) = %lu\n"
-"[Info] libdvdsize      = %lu\n"
-msgstr ""
-
-#: vobcopy.c:1461
-#, c-format
-msgid ""
-"\n"
-"[Info]  Disk free: %.0f MB"
-msgstr ""
-
-#: vobcopy.c:1462
-#, c-format
-msgid ""
-"\n"
-"[Info]  Vobs size: %.0f MB"
-msgstr ""
-
-#: vobcopy.c:1464
-#, c-format
-msgid ""
-"\n"
-"[Error] Hmm, better change to a dir with enough space left or call with -f "
-"(force) \n"
-msgstr ""
-
-#: vobcopy.c:1468 vobcopy.c:1469
-#, c-format
-msgid "[Error] Hmm, statfs (statvfs) seems not to work on that directory. \n"
-msgstr ""
-
-#: vobcopy.c:1470
-#, c-format
-msgid ""
-"[Hint] Nevertheless, do you want vobcopy to continue [y] or do you want to "
-"check for \n"
-msgstr ""
-
-#: vobcopy.c:1471
-#, c-format
-msgid "[Hint] enough space first [q]?\n"
-msgstr ""
-
-#: vobcopy.c:1481
-#, c-format
-msgid "[Info] y pressed - force write\n"
-msgstr ""
-
-#: vobcopy.c:1487
-#, c-format
-msgid "[Info] n/q pressed\n"
-msgstr ""
-
-#: vobcopy.c:1493
-#, c-format
-msgid "[Error] Please choose [y] to continue or [n] to quit\n"
-msgstr ""
-
-#: vobcopy.c:1518
-#, c-format
-msgid ""
-"\n"
-"[Info] Your-name for the dvd: %s\n"
-msgstr ""
-
-#: vobcopy.c:1534
-#, c-format
-msgid "[Info] Free space for -o dir: %.0f\n"
-msgstr ""
-
-#: vobcopy.c:1563
-#, c-format
-msgid "[Info] Free space for -%i dir: %.0f\n"
-msgstr ""
-
-#: vobcopy.c:1579
-#, c-format
-msgid "[Info] Taken max_filesize_in_blocks(2GB version): %.0f\n"
-msgstr ""
-
-#: vobcopy.c:1587
-#, c-format
-msgid "[Info] Taken max_filesize_in_blocks(lfs version): %.0f\n"
-msgstr ""
-
-#: vobcopy.c:1627
-#, c-format
-msgid ""
-"\n"
-"[Hint] please choose [o]verwrite or [q]uit the next time ;-)\n"
-msgstr ""
-
-#: vobcopy.c:1643
-#, c-format
-msgid "[Error] Seems your platter is full...\n"
-msgstr ""
-
-#: vobcopy.c:1646
-#, c-format
-msgid ""
-"\n"
-"[Error] File '%s' already exists, [o]verwrite, [a]ppend, [q]uit? \n"
-msgstr ""
-
-#: vobcopy.c:1683
-#, c-format
-msgid "[Info] User chose append\n"
-msgstr ""
-
-#: vobcopy.c:1694
-#, c-format
-msgid ""
-"\n"
-"[Hint] Please choose [o]verwrite, [a]ppend, [q]uit the next time ;-)\n"
-msgstr ""
-
-#: vobcopy.c:1746
-#, c-format
-msgid ""
-"\n"
-"[Error] Write() error\n"
-msgstr ""
-
-#: vobcopy.c:1762
-#, c-format
-msgid "%4.0fMB of %4.0fMB written (%.0f %%)\r"
-msgstr ""
-
-#: vobcopy.c:1778
-#, c-format
-msgid "%4.0fMB of %4.0fMB written ( 100.0 %%)\r"
-msgstr ""
-
-#: vobcopy.c:1786
-#, c-format
-msgid "[Info] max_filesize_in_blocks %8.0f \n"
-msgstr ""
-
-#: vobcopy.c:1787
-#, c-format
-msgid "[Info] offset at the end %8.0f \n"
-msgstr ""
-
-#: vobcopy.c:1788
-#, c-format
-msgid "[Info] file_size_in_blocks %8.0f \n"
-msgstr ""
-
-#: vobcopy.c:1803
-#, c-format
-msgid ""
-"\n"
-"[Error] File size (%.0f) of %s differs largely from that on dvd, therefore "
-"keeps it's .partial\n"
-msgstr ""
-
-#: vobcopy.c:1827
-#, c-format
-msgid "[Info] Single file size (of copied file %s ) %.0f\n"
-msgstr ""
-
-#: vobcopy.c:1828
-#, c-format
-msgid "[Info] Cumulated size %.0f\n"
-msgstr ""
-
-#: vobcopy.c:1832
-#, c-format
-msgid "[Info] Successfully copied file %s\n"
-msgstr ""
-
-#: vobcopy.c:1839
-#, c-format
-msgid "[Info] # of separate files: %i\n"
-msgstr ""
-
-#: vobcopy.c:1849
-#, c-format
-msgid ""
-"\n"
-"[Info] Copying finished! Let's see if the sizes match (roughly)\n"
-msgstr ""
-
-#: vobcopy.c:1850
-#, c-format
-msgid "[Info] Combined size of title-vobs: %.0f (%.0f MB)\n"
-msgstr ""
-
-#: vobcopy.c:1851
-#, c-format
-msgid "[Info] Copied size (size on disk):  %.0f (%.0f MB)\n"
-msgstr ""
-
-#: vobcopy.c:1855
-#, c-format
-msgid "[Error] Hmm, the sizes differ by more than %d\n"
-msgstr ""
-
-#: vobcopy.c:1856
-#, c-format
-msgid "[Hint] Take a look with MPlayer if the output is ok\n"
-msgstr ""
-
-#: vobcopy.c:1860
-#, c-format
-msgid "[Info] Everything seems to be fine, the sizes match pretty good ;-)\n"
-msgstr ""
-
-#: vobcopy.c:1861
-#, c-format
-msgid "[Hint] Have a lot of fun!\n"
-msgstr ""
-
-#: vobcopy.c:1910 vobcopy.c:1954
-#, c-format
-msgid "[Info] Used the linux statfs\n"
-msgstr ""
-
-#: vobcopy.c:1914 vobcopy.c:1958
-#, c-format
-msgid "[Info] Used statvfs\n"
-msgstr ""
-
-#: vobcopy.c:1928
-#, c-format
-msgid "[Info] In freespace_getter:for %s : %.0f free\n"
-msgstr ""
-
-#: vobcopy.c:1929
-#, c-format
-msgid "[Info] In freespace_getter:bavail %ld * bsize %ld = above\n"
-msgstr ""
-
-#: vobcopy.c:1972
-#, c-format
-msgid "[Info] In usedspace_getter:for %s : %.0f used\n"
-msgstr ""
-
-#: vobcopy.c:1973
-#, c-format
-msgid "[Info] In usedspace_getter:part1 %ld, part2 %ld\n"
-msgstr ""
-
-#: vobcopy.c:2012
-#, c-format
-msgid ""
-"\n"
-"[Info] Outputting to %s"
-msgstr ""
-
-#: vobcopy.c:2044
-#, c-format
-msgid "[-l (large-file support for files > 2GB)] \n"
-msgstr ""
-
-#: vobcopy.c:2075
-#, c-format
-msgid "[Error] Could not remove old filename: %s \n"
-msgstr ""
-
-#: vobcopy.c:2076
-#, c-format
-msgid "[Hint] This: %s is a hardlink to %s. Dunno what to do... \n"
-msgstr ""
-
-#: vobcopy.c:2085
-#, c-format
-msgid "[Error] File %s already exists! Gonna name the new one %s.dupe \n"
-msgstr ""
-
-#: vobcopy.c:2117
-#, c-format
-msgid "[Error] The directory %s already exists!\n"
-msgstr ""
-
-#: vobcopy.c:2118
-#, c-format
-msgid "[Hint] You can either [c]ontinue writing to it or you can [q]uit: "
-msgstr ""
-
-#: vobcopy.c:2136
-#, c-format
-msgid ""
-"\n"
-"[Hint] please choose [c]ontinue or [q]uit the next time ;-)\n"
-msgstr ""
-
-#: vobcopy.c:2143
-#, c-format
-msgid ""
-"[Error] Creating of directory %s\n"
-" failed! \n"
-msgstr ""
-
-#: vobcopy.c:2168
-#, c-format
-msgid ""
-"\n"
-"[Info] Timer expired - shooting myself in the head.\n"
-msgstr ""
-
-#: vobcopy.c:2174
-#, c-format
-msgid ""
-"\n"
-"[Info] Terminate signal received, exiting.\n"
-msgstr ""
